app/modules/enduser_setup.c

```diff
@@ -490,8 +520,15 @@ static void enduser_setup_http_urldecode(char *dst, const char *src, int src_len
       }
       *dst++ = c;
     }
+    if ((dst - dst_start) >= dst_len - 1)
```

Comment:
Don't we need to have the test against `dst_len` as part of the outer loop? Right now it looks like it can overrun as long as it's not in an escape sequence.